allennlp/nn/decoding/decoder_trainers/expected_risk_minimization.py

```diff
@@ -82,10 +88,39 @@ def _get_finished_states(self,
                 else:
                     next_states.append(next_state)
 
-            states = self._prune_beam(next_states)
+            states = self._prune_beam(next_states, self._beam_size, False)
             num_steps += 1
+        if self._max_num_finished_states is not None:
+            finished_states = self._prune_beam(finished_states, self._max_num_finished_states, True)
         return finished_states
 
+    @staticmethod
```

Comment:
This doesn't need to happen now, but this method could probably just be put into `nn.decoding.util`, as lots of places want to prune a grouped beam.

Reply:
That's a good idea. Added a TODO for now.